### DISCUSSION TITLE
Is it expected that children of `<Stepper.Step>` are mounted twice?

### DISCUSSION THREAD

**Discussion by pkalita-lbl** [mantinedev/mantine#2754] [Q&A] [answered]
Here's a small example that renders a `<Stepper>` with two `<Stepper.Step>`s:

https://codesandbox.io/s/flamboyant-poincare-porzpd?file=/src/App.js

The children of the `<Stepper.Step>`s have a `useEffect` hook with an empty dependency array. The idea being that some side-effect of moving to that step can happen in the `useEffect` callback. However the `useEffect` callback is consistently called twice when moving to the given step. I would have expected it to only be called once. Is the observed behavior expected?

**Comment by rtivital** [comment, marked as answer]
useEffect is always called twice when react strict mode is enabled. If you disable it, useEffect is called once – https://codesandbox.io/s/sharp-kapitsa-nby8k2

**Reply by pkalita-lbl** [reply]
Thanks, you're absolutely right. I guess it's [this React 18 behavior](https://reactjs.org/docs/strict-mode.html#ensuring-reusable-state) that I hadn't encountered yet.
